pkg/controller/s3/bucket/websiteConfig.go

```diff
@@ -139,28 +139,37 @@ func GenerateWebsiteConfiguration(config *v1beta1.WebsiteConfiguration) *awss3.W
 	if config.RedirectAllRequestsTo != nil {
 		wi.RedirectAllRequestsTo = &awss3.RedirectAllRequestsTo{
 			HostName: awsclient.String(config.RedirectAllRequestsTo.HostName),
-			Protocol: awss3.Protocol(config.RedirectAllRequestsTo.Protocol),
+		}
+		if config.RedirectAllRequestsTo.Protocol != "" {
+			wi.RedirectAllRequestsTo.Protocol = awss3.Protocol(config.RedirectAllRequestsTo.Protocol)
 		}
 	}
-	wi.RoutingRules = make([]awss3.RoutingRule, len(config.RoutingRules))
-	for i, rule := range config.RoutingRules {
-		rr := awss3.RoutingRule{
-			Redirect: &awss3.Redirect{
+	if config.RoutingRules != nil {
```

Comment:
```suggestion
	if len(config.RoutingRules) != 0 {
```

This is usually a safer check for slices.

Reply:
done